hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactBooleanFieldTest.java

```diff
@@ -0,0 +1,205 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl.compact;
+
+import com.hazelcast.config.CompactSerializationConfig;
+import com.hazelcast.config.SerializationConfig;
+import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder;
+import com.hazelcast.nio.serialization.compact.CompactReader;
+import com.hazelcast.nio.serialization.compact.CompactSerializer;
+import com.hazelcast.nio.serialization.compact.CompactWriter;
+import com.hazelcast.test.HazelcastParametrizedRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static com.hazelcast.internal.serialization.impl.compact.CompactTestUtil.createSerializationService;
+
+@RunWith(HazelcastParametrizedRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class CompactBooleanFieldTest {
+    private final SchemaService schemaService = CompactTestUtil.createInMemorySchemaService();
+
+    @Parameterized.Parameter
+    public int itemCount;
+
+
+    @Parameterized.Parameters(name = "itemCount:{0}")
+    public static Object[] parameters() {
+        return new Object[]{0, 1, 8, 10, 100, 1000};
+    }
+
+    private static boolean[] getBooleans(int count) {
+        boolean[] booleans = new boolean[count];
+        Random r = new Random();
+        for (int i = 0; i < count; i++) {
+            booleans[i] = r.nextInt(10) % 2 == 0;
+        }
+        return booleans;
+    }
+
+    @Test
+    public void testBooleanArrayWithCustomSerializer() {
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.addSerializer(new BoolArrayDTOSerializer());
+        SerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
+        boolean[] bools = getBooleans(itemCount);
+        BoolArrayDTO expected = new BoolArrayDTO(bools);
+
+        Data data = serializationService.toData(expected);
+        BoolArrayDTO actual = serializationService.toObject(data);
```

Comment:
if there is error in `toData` and `toObject` and they cancel each other, then the test passes but could not catch the problem. I would expect actual binary comparison at this test?

Similar comment for other tests as well.

Reply:
For which tests should we have a binary check? The thing you mention can happen in all compact tests IMO. @mdumandag WDYT?

Reply:
I don't think it is practical to do binary comparison in every test, but maybe we can add something for Compact to `BinaryCompatibilityTest`, similar to other serialization mechanisms

Reply:
I will add MainDTO to `BinaryCompatibilityTest` (that is to ReferenceObjects)

Reply:
@srknzl did you add to BinaryCompatibilityTest in this PR? I could not see it.

Reply:
Pushed recently.